pennylane/ops/op_math/sprod.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the SProd class which contains logic for
+computing the scalar product of operations.
+"""
+import pennylane as qml
+from .symbolicop import SymbolicOp
+
+
+def s_prod(scalar, operator, do_queue=True, id=None):
+    r"""Construct an operator which is the scalar product of the
+     given scalar and operator provided.
+
+    Args:
+        scalar (float or complex): the scale factor being multiplied to the operator.
+        operator (~.operation.Operator): the operator which will get scaled.
+
+    Keyword Args:
+        do_queue (bool): determines if the scalar product operator will be queued
+            (currently not supported). Default is True.
+        id (str or None): id for the scalar product operator. Default is None.
+
+    Returns:
+        ~ops.op_math.SProd: the operator representing the scalar product.
+
+    ..seealso:: :class:`~.ops.op_math.SProd`
+
+    **Example**
+
+    >>> sprod_op = s_prod(2.0, qml.PauliX(0))
+    >>> sprod_op
+    2.0*(PauliX(wires=[0]))
+    >>> sprod_op.matrix()
+    array([[ 0., 2.],
+           [ 2., 0.]])
+    """
+    return SProd(scalar, operator, do_queue=do_queue, id=id)
+
+
+def _sprod(mat, scalar, dtype=None, cast_like=None):
```

Comment:
I don't think we need to extract this code to a private helper function.  The `dtype` and `cast_like` keywords are unnecessary since we don't use them.  Get rid of the keywords, and the function is just `scalar*mat`.  This just leads to extra indirection and source code complexity.

If the `sparse_matrix` and `matrix` methods get more complex over time, we can extract the code, but that is a future problem.  For now, we should just do `scalar*mat` in the actual class methods.

Reply:
Removed!